oslo/transformers/tasks/data_bart_pretraining.py

```diff
@@ -102,122 +142,149 @@ def __init__(
-        def _possion_segmentation(num_noise_tokens):
-            segment_lengths = []
-            while sum(segment_lengths) < num_noise_tokens:
-                span_length = np.random.poisson(lam=self.possion_lambda)
-                segment_lengths.append(span_length)
-
-            difference = sum(segment_lengths) - num_noise_tokens
-            segment_lengths[-1] = segment_lengths[-1] - difference
-            segment_lengths.sort(reverse=True)
-            return segment_lengths
-
-        temp_ids = input_ids
-        while len(temp_ids) >= length:
-            temp_ids = input_ids[:]
-            noise_span_lengths = _possion_segmentation(num_noise_tokens)
-
-            for noise_span_length in noise_span_lengths:
-                max_idx = len(temp_ids) - noise_span_length + 1
-                # get start index of mask span
-                start_indices = self.get_start_indices[max_idx]
-                for start_idx in start_indices:
-                    if (
-                        self.mask_token_id
-                        in temp_ids[start_idx : start_idx + noise_span_length]
-                    ):
-                        continue
-                    else:
-                        temp_ids = (
-                            temp_ids[:start_idx]
-                            + [self.mask_token_id]
-                            + temp_ids[start_idx + noise_span_length :]
-                        )
-                        # rotate start indices
-                        self.get_start_indices[max_idx] = np.roll(start_indices, 1)
-                        break
-
-        input_ids = temp_ids
-
-        return input_ids
-
-    def _sentence_permutation(self, input_ids: List[int]) -> List[int]:
-        ref_tokens = self.tokenizer.convert_ids_to_tokens(input_ids)
-
-        split_sentences = []
-        split_points = [
-            idx for idx, token in enumerate(ref_tokens) if token in (".", "Ġ.")
-        ]
-
-        if split_points:
-            prev_point = 0
-            for split_point in split_points:
-                split_point += 1
-                split_sentences.append(input_ids[prev_point:split_point])
-                prev_point = split_point
-            split_sentences.append(input_ids[prev_point:])
-
-            random.shuffle(split_sentences)
-
-            input_ids = []
-            for split_sentence in split_sentences:
-                input_ids.extend(split_sentence)
-
-        return input_ids
-
-    def _prepare_decoder_inputs_from_labels(self, batch):
-        # Shift input ids one token to the right
-        shifted_input_ids = batch["labels"].new_zeros(batch["labels"].shape)
-        shifted_input_ids[:, 1:] = batch["labels"][:, :-1].clone()
-        shifted_input_ids[:, 0] = self.decoder_start_token_id
-
-        shifted_input_ids.masked_fill_(
-            shifted_input_ids == self.label_pad_token_id,
-            self.pad_token_id,
-        )
+    def permute_sentences(self, input_ids):
```

Comment:
permute sentence function - 

Check the end point of the sentence based on pad_token and mix the sentences according to the percentage of permit_sentence_ratio.